aten/src/THC/generic/THCTensorMathBlas.cu

```diff
@@ -687,8 +687,10 @@ void THCTensor_(baddbmm)(THCState *state, THCTensor *result, scalar_t beta, THCT
         THCTensor_(data)(state, result_) + i * result_->stride(0), ldc);
   }
 #else
+#ifndef __HIP_PLATFORM_HCC__
   cudaDeviceProp* prop = THCState_getCurrentDeviceProperties(state);
   if (prop->major >= 5){
```

Comment:
does hip also need to check the major version? https://github.com/pytorch/pytorch/blob/7e2b074219fad6d2b09b379423e83b2295b29df2/caffe2/core/common_gpu.h#L81-L86

Reply:
gemm_ex is supported on all ROCm supported gfx architectures, performance for mixed precision will be different based on architecture characteristics

Reply:
I disagree. Only for gfx900 will it not have faster intrinsics so performance will be comparable to sgemm. However, you still profit from slightly reduced memory bandwidth usage and gain accuracy over straight hgemm. From gfx906 on, it'll be fast and accurate. Debugging arch-dependent accuracy issues doesn't sound like fun to me...

Reply:
@iotamudelta In Caffe2, I used major version 3 (gfx9) which I thought is practically the min. version we support.

Reply:
@rohithkrn @iotamudelta Could you confirm whether this is supported in all archs or only in major version >= 3?

Reply:
@bddppq rocBLAS gemm_ex is supported on all uarchs that ROCm supports (so even below gfx900 / below major version 3). What @rohithkrn is pointing out is that we never did any tests on or even compilations for < 3.

Reply:
Ok I see. In this case, @rohithkrn could you change caffe2 to also not check the major version? (in a separate PR)

Reply:
will do